tsl/src/compression/compression.c

```diff
@@ -995,6 +948,34 @@ row_compressor_append_sorted_rows(RowCompressor *row_compressor, Tuplesortstate
 	ExecDropSingleTupleTableSlot(slot);
 }
 
+static void
+row_compressor_process_ordered_slot(RowCompressor *row_compressor, TupleTableSlot *slot,
+									CommandId mycid)
+{
+	MemoryContext old_ctx;
+	slot_getallattrs(slot);
+	old_ctx = MemoryContextSwitchTo(row_compressor->per_row_ctx);
+	if (row_compressor->first_iteration)
+	{
+		row_compressor_update_group(row_compressor, slot);
+		row_compressor->first_iteration = false;
+	}
+	bool changed_groups = row_compressor_new_row_is_in_new_group(row_compressor, slot);
+	bool compressed_row_is_full =
+		row_compressor->rows_compressed_into_current_value >= MAX_ROWS_PER_COMPRESSION;
+	if (compressed_row_is_full || changed_groups)
+	{
+		if (row_compressor->rows_compressed_into_current_value > 0)
```

Comment:
note: this line seems to be a precondition to run `row_compressor_flush` ; I think it could make sense to move it to the beginning of that method

Reply:
Did not want to change the core logic though what you are saying makes sense.